pkg/resourceutils/scaleDown.go

```diff
@@ -0,0 +1,270 @@
+			APIVersion: cronJobGVK.Group + "/" + cronJobGVK.Version},
+	}
+	_, err = dynamicClient.Update(context.TODO(), &object, *opts, "")
+	if err != nil {
+		return fmt.Errorf("unable to update resource %v %v/%v: %v", strings.ToLower(cronJobGVK.Kind), object.GetNamespace(), object.GetName(), err)
+	}
+	fmt.Printf("successfully updated resource %v %v/%v \n", strings.ToLower(cronJobGVK.Kind), object.GetNamespace(), object.GetName())
+	return nil
+}
+
+func scaleDownCRDResource(dynamicClient k8sdynamic.ResourceInterface, object unstructured.Unstructured, crd v1alpha1.ApplicationResource, config *rest.Config) error {
+	content := object.UnstructuredContent()
+	annotations, found, err := unstructured.NestedStringMap(content, "metadata", "annotations")
+	if err != nil {
+		return err
+	}
+	if !found {
+		annotations = make(map[string]string)
+	}
+	if crd.SuspendOptions.Path != "" {
+		crd.NestedSuspendOptions = append(crd.NestedSuspendOptions, crd.SuspendOptions)
+	}
+	if len(crd.NestedSuspendOptions) == 0 {
+		return nil
+	}
+	for _, suspend := range crd.NestedSuspendOptions {
+		specPath := strings.Split(suspend.Path, ".")
+		if len(specPath) > 1 {
+			var val string
+			SuspendAnnotationValue, suspendAnnotationIsPresent := annotations[migration.StorkAnnotationPrefix+suspend.Path]
+			var disableVersion interface{}
+			if suspend.Type == "int" {
+				currentValue, found, err := unstructured.NestedInt64(content, specPath...)
+				if err != nil || !found {
+					return fmt.Errorf("unable to find suspend path, err: %v", err)
+				}
+				disableVersion = int64(0)
+				if currentValue == 0 && suspendAnnotationIsPresent {
+					// suspendAnnotation has value set as {currVal + "," + suspend.Value}
+					// we need to extract only the currVal from it
+					annotationValue := strings.Split(SuspendAnnotationValue, ",")[0]
+					intValue, err := strconv.ParseInt(annotationValue, 10, 64)
+					if err != nil {
+						return err
+					}
+					if intValue > 0 {
+						// If the actual value is 0 and suspend annotation value is more than 0,
+						// then don't override the annotation.
+						currentValue = intValue
+					}
+				}
+				val = fmt.Sprintf("%v", currentValue)
+			} else if suspend.Type == "string" {
+				currentValue, _, err := unstructured.NestedString(content, specPath...)
+				if err != nil {
+					return fmt.Errorf("unable to find suspend path, err: %v", err)
+				}
+				disableVersion = suspend.Value
+				if currentValue == suspend.Value && suspendAnnotationIsPresent {
+					annotationValue := strings.Split(SuspendAnnotationValue, ",")[0]
+					if annotationValue != "" {
+						// If the actual value is equal to suspend.Value (i.e. 0 value) and suspend path annotation value is not an empty string,
+						// then don't override the annotation
+						currentValue = annotationValue
+					}
+				}
+				val = currentValue
+			} else if suspend.Type == "bool" {
+				if val, err := strconv.ParseBool(suspend.Value); err != nil {
+					disableVersion = true
```

Comment:
Shouldn't we log that there was a parsing error here?

Reply:
Makes sense. Will add a log statement